docs/docs/context.md

```diff
@@ -198,14 +202,6 @@ MediaQuery.childContextTypes = {
 };
 ```
 
-## When not to use context
-
-Just as global variables are best avoided when writing clear code, you should avoid using context in most cases. In particular, think twice before using it to "save typing" and using it instead of passing explicit props.
-
-The best use cases for context are for implicitly passing down the logged-in user, the current language, or theme information. All of these might otherwise be true globals, but context lets you scope them to a single React subtree.
-
-Do not use context to pass your model data through components. Threading your data through the tree explicitly is much easier to understand. Using context makes your components more coupled and less reusable, because they behave differently depending on where they're rendered.
-
-## Known limitations
+## Known Limitations
 
 If a context value provided by a component changes, descendants that use that value won't update if an intermediate parent returns `false` from `shouldComponentUpdate`. See issue [#2517](https://github.com/facebook/react/issues/2517) for more details.
```

Comment:
In general I feel like not enough time is spent explaining that this is an experimental API and it is likely to break in the future. Unlike props it is likely you'll have to change every single component that he uses context when we change its API. 

Also even though you describe how updates work they are just broken. The limitation in the last sentence is very severe because it is completely out of control of the components using context. So essentially you can consider updates to context broken, abd we should recommend treating it as unchanging because it's the only way to use it safely. You can still pass your own subscription system through it.

This is a really good article about it. I'm not sold on its conclusion ("wrap everything with MobX") because MobX wraps all native data structures into its own, and has its share of issues related to that, but the rest of the article highlights pitfalls of context very well:

https://medium.com/@mweststrate/how-to-safely-use-react-context-b7e343eff076